hibernate-core/src/main/java/org/hibernate/query/hql/internal/SemanticQueryBuilder.java

```diff
@@ -3603,8 +3606,9 @@ public Object visitStrFunction(HqlParser.StrFunctionContext ctx) {
 
 		final SqmExpression<?> argument = (SqmExpression<?>) ctx.predicate().accept( this );
 
-		return getFunctionDescriptor("every").generateSqmExpression(
```

Comment:
What I meant in https://github.com/hibernate/hibernate-orm/pull/4075#discussion_r660577299 is that you add the following:

```
if ( argument instanceof SqmSubQuery<?> && ctx.filterClause() != null ) {
    throw new SemanticException( "Quantified expression can not have a filter clause!" );
}
```